src/controller/artifact/controller.go

```diff
@@ -227,6 +227,7 @@ func (c *controller) ensureArtifact(ctx context.Context, repository, digest stri
 		if !errors.IsConflictErr(err) {
 			return false, nil, err
 		}
+		log.Debugf("failed to create artifact %d: %v", art.ID, err)
```

Comment:
The var `art` is assigned in the first line of this function, it should be nil if the artifact is not found, so here print this may cause the panic.

Reply:
Thanks for the catch! I also added a test case in the unit test for this code path.